src/libstd/sys/unix/process.rs

```diff
@@ -311,6 +312,18 @@ impl Process {
         if !envp.is_null() {
             *sys::os::environ() = envp as *const _;
         }
+
+        // Reset signal handling so the child process starts in a
+        // standardized state. We ignore SIGPIPE, and signal-handling
+        // libraries often set a mask; both of these get inherited,
+        // which most UNIX programs don't expect.
+        let mut set: c::sigset_t = mem::uninitialized();
+        if c::sigemptyset(&mut set) != 0 ||
+           c::pthread_sigmask(c::SIG_SETMASK, &set, ptr::null_mut()) != 0 ||
+           c::signal(libc::SIGPIPE, c::SIG_DFL) == c::SIG_ERR {
+            fail(&mut output);
+        }
```

Comment:
> pthread_sigmask is a strict superset of sigprocmask: it's defined to do the same thing if the process is single-threaded, and has well-defined behavior if the process is multithreaded, so I've just been making myself forget that sigprocmask exists. It's also one fewer thing to bind, which also reduces the temptation for a future commit to call it in a maybe-multithreaded context (e.g., if you wanted to mask signals around the call to fork, as libc system does).

Sounds good to me

> Just the antecedent of "both of these"?

Ah I think I just didn't reread after learning more about what's going on here, I think it's fine.

---

So just to check I poked around at some other libraries to see what they do. In libuv at least, nothing with signals is handled in terms of forking, but they also don't by default disable SIGPIPE. I couldn't make heads or tails of what Ruby does, but it didn't look like Python did much with signal masks or forking despite ignoring SIGPIPE like we are.

Given this information, it strikes me as somewhat odd that we're resetting the signal mask? Shouldn't that be up to the application instead of the library here instead? For example this makes it so it's impossible to spawn a process with a signal mask in place through the `Command` API. This info also makes me uneasy about ignoring `SIGPIPE` by default, but I think that setting it to `SIG_DFL` here is fine.